prae/tests/ensure.rs

```diff
@@ -19,10 +26,7 @@ fn construction_succeeds_for_valid_data() {
 #[test]
 fn mutation_fails_for_invalid_data() {
     let mut un = Username::new("user").unwrap();
-    assert_eq!(
-        un.try_mutate(|u| *u = "".to_owned()).unwrap_err(),
-        prae::ValidationError
-    );
+    let _error: prae::ConstructionError<String> = un.try_mutate(|u| *u = "".to_owned()).unwrap_err();
```

Comment:
Why not `assert_matches!`?

Reply:
I'll put assert_matches back in. I was having trouble getting the match to compile using the generic param, but all I need is a turbo fish.

Reply:
The `assert_matches!` are restored.